src/commands/lint/mod.rs

```diff
@@ -0,0 +1,118 @@
+pub use args::*;
+#[cfg(feature = "rustup")]
+use serde::Deserialize;
+
+#[cfg(feature = "rustup")]
+use crate::external_cli::Package;
+
+mod args;
+
+/// Represents the contents of `rust-toolchain.toml`.
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct RustToolchain {
+    toolchain: Toolchain,
+}
+
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct Toolchain {
+    channel: String,
+}
+
+/// Runs `bevy_lint`, if it is installed, with the given arguments.
+///
+/// Calling `lint(vec!["--workspace"])` is equivalent to calling `bevy_lint --workspace` in the
+/// terminal. This will run [`find_bevy_lint()`] to locate `bevy_lint`.
+pub fn lint(args: &mut LintArgs) -> anyhow::Result<()> {
+    use anyhow::ensure;
+
+    #[cfg(feature = "web")]
+    use crate::commands::lint::args::LintSubcommands;
+    use crate::{bin_target::select_run_binary, config::CliConfig, external_cli::cargo};
+
+    let auto_install = args.auto_install();
+
+    let metadata = cargo::metadata::metadata()?;
+
+    let mut bin_target = select_run_binary(
+        &metadata,
+        args.cargo_args.package_args.package.as_deref(),
+        args.cargo_args.target_args.bin.as_deref(),
+        args.cargo_args.target_args.example.as_deref(),
+        args.target().as_deref(),
+        args.profile(),
+    )?;
+
+    let config = CliConfig::for_package(
+        &metadata,
+        bin_target.package,
+        args.is_web(),
+        args.is_release(),
+    )?;
+
+    args.apply_config(&config);
+    // Update the artifact directory based on the config, e.g. in case the `target` changed
+    bin_target.update_artifact_directory(
+        &metadata.target_directory,
+        args.target().as_deref(),
+        args.profile(),
+        args.cargo_args.target_args.example.is_some(),
+    );
+
+    #[allow(
+        unused_mut,
+        reason = "This only needs to be mutable when the web feature is enabled"
+    )]
```

Comment:
```suggestion
    #[cfg_attr(
        not(feature = "web"),
        expect(
            unused_mut,
            reason = "this is only mutated when the `web` feature is enabled"
        )
    )]
```

(Nit) It's a bit more verbose, but this is stricter and will help us catch if this ever needs to be changed